lld/COFF/SymbolTable.cpp

```diff
@@ -74,13 +77,18 @@ void SymbolTable::addFile(InputFile *file) {
   }
 
   MachineTypes mt = file->getMachineType();
+  if (!compatibleMachineType(ctx, mt)) {
+    if (isArm64EC(mt))
+      error(toString(file) + ": incompatible machine type " + machineToStr(mt) +
```

Comment:
Not a fan of this error message: it seems odd that if the user asked for the machine type to be inferred that the linker reports that something is incompatible. Maybe say that it is ambiguous or can't be decided/inferred?

Reply:
But isn't it possible to hit this case too, e.g. if you're linking an i386 module, but you're hitting an arm64ec input file?

Reply:
I.e., would it make sense to try to disambiguate these two cases here, for the sake of getting more easily understandable error messages?